eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Data.hs

```diff
@@ -316,50 +345,53 @@ instance
     where
       decodeShelley =
         decode
-          ( Ann (Emit AuxiliaryDataRaw)
+          ( Ann (Emit AlonzoTxAuxDataRaw)
               <*! Ann From
               <*! Ann (Emit StrictSeq.empty)
+              <*! Ann (Emit Map.empty)
           )
       decodeShelleyMA =
         decode
-          ( Ann (RecD AuxiliaryDataRaw)
+          ( Ann (RecD AlonzoTxAuxDataRaw)
               <*! Ann From
               <*! D
-                ( sequence
-                    <$> decodeStrictSeq
-                      (fmap TimelockScript <$> fromCBOR)
-                )
+                (sequence <$> decodeStrictSeq fromCBOR)
+              <*! Ann (Emit Map.empty)
           )
       decodeAlonzo =
         decode $
           TagD 259 $
             SparseKeyed "AuxiliaryData" (pure emptyAuxData) auxDataField []
 
-      auxDataField :: Word -> Field (Annotator (AuxiliaryDataRaw era))
-      auxDataField 0 = fieldA (\x ad -> ad {txMD' = x}) From
+      addPlutusScripts lang scripts ad =
+        case NE.nonEmpty scripts of
+          Nothing -> ad
+          Just neScripts ->
+            -- Avoid leaks by , since non empty list is lazy.
```

Comment:
```suggestion
            -- Avoid leaks by deepseq, since non empty list is lazy.
```

is that what you meant? :point_up:

Reply:
Yes, that was exactly what I was trying to type in there, but for some mysterious reason it got typed in my head only. :smile: